src/components/HoverableButton/HoverableButton.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { TreeviewItemHoverBtn } from '../Treeview/treeviewItemHoverBtn';
+import { IHoverOverBtn } from '../Treeview';
+import './HoverableButton.scss';
+
+export interface IHoverableButtonsProps {
+    id: string;
+    hoverOverBtn: Array<IHoverOverBtn>;
+}
+
+export interface IHoverableButtonsState {
+    hover: boolean;
+}
+
+const addHoverableButtons = (hoverableProps: IHoverableButtonsProps) => <P extends {} >( Comp: React.ComponentClass<P> | React.StatelessComponent<P>): React.ComponentClass<P> => {
+    return class HoverableButtons extends React.Component<P, IHoverableButtonsState> {
+        public constructor(props: P) {
+            super(props);
+     
+            this.state = {
+                hover: false
+            };
+        }
+    
+        render(): JSX.Element {
+            if (!hoverableProps.hoverOverBtn) {
+                return <Comp {...this.props} />;
+            }
+
+            return <span className="hoverable-items__container" onMouseEnter={this._onItemHover.bind(this)} onMouseLeave={this._onItemLeaveHover.bind(this)}>
```

Comment:
can we use autobind?

Reply:
@vinkobedek unfortunately no, when i tried it said that decorators can't be used in this context